Persist ffis data

jakekreider opened the pull request:
### Ticket #13
Closes #13 

## Description
Per ticket requirements, defines a Lambda that listens to the prepared grants data bucket and upserts FFIS data by grant ID. 

After discussion, the only field we're updating (at this time) is "Bill", which at times has also been called "Act", "Appropriation Act", and "Program".

## Testing
* Upload test JSON with `awslocal s3 cp cmd/PersistFFISData/fixtures/standard.json s3://grants-ingest-grantssourcedata-000000000000-us-west-2/sources/2023/04/2sd/ffis/v1.json`. You can then tail the logs, or wait a few seconds and check DynamoDB with `awslocal dynamodb scan --table-name grants-ingest-prepareddata`.
* If you have a Grants.gov DB handy, you can upload that to pre-seed the DB. `awslocal s3 cp GrantsDBExtract.xml s3://grants-ingest-grantssourcedata-000000000000-us-west-2/sources/2023/04/11/grants.gov/extract.xml`

### Automated and Unit Tests
- [x] Added Unit tests

### Manual tests for Reviewer
- [ ] Added steps to test feature/functionality manually

## Checklist
- [ ] Provided ticket and description
- [ ] Provided testing information
- [ ] Provided adequate test coverage for all new code
- [ ] Added PR reviewers

jakekreider commented on cmd/PersistFFISData/main.go:
```diff
@@ -0,0 +1,63 @@
+var (
+	env    Environment
+	logger log.Logger
+	// sendMetric       = ddHelpers.NewMetricSender("PersistFFISData", "source:grants.gov")
```
I've been copying these in from the original pattern/template - can I remove?

TylerHendrickson replied:
Normally, yes – no expectation to keep unused code/comments for the sake of convention – but in this case, I think it would be good to emit an `opportunity.saved` metric after successfully saving to DDB. I dropped a suggestion in the `handleS3Event()` function accordingly, although it could also be sent from `UpdateOpportunity()` if you think that's more appropriate. The idea is to have this mirror the `opportunity.saved` metric [emitted by `PersistGrantsGovXMLDB`](https://github.com/usdigitalresponse/grants-ingest/blob/main/cmd/PersistGrantsGovXMLDB/handler.go#LL96C2-L96C36), allowing us to track DB updates across both sources.

Following that, this should be uncommented and the `source` tag should be updated to reflect FFIS rather than Grants.gov:
```suggestion
	sendMetric = ddHelpers.NewMetricSender("PersistFFISData", "source:ffis")
```

TylerHendrickson commented on cmd/PersistFFISData/handler.go:
```diff
@@ -0,0 +1,68 @@
+	if err != nil {
+		return err
+	}
+	err = UpdateOpportunity(ctx, dbapi, env.DestinationTable, opportunity(ffisData))
+	return err
```
```suggestion
	err = UpdateOpportunity(ctx, dbapi, env.DestinationTable, opportunity(ffisData))
	if err == nil {
		sendMetric("opportunity.saved", 1)
	}
	return err
```

TylerHendrickson reviewed (changes requested):
Looks great! I addressed your comment and included two corresponding suggestions, but other than that I think this looks good to go.

jakekreider commented:
> Looks great! I addressed your comment and included two corresponding suggestions, but other than that I think this looks good to go.

Cool, appreciate it!
